storagenode/contact/endpoint.go

```diff
@@ -42,6 +52,22 @@ func (endpoint *Endpoint) Ping(ctx context.Context, req *pb.ContactPingRequest)
 		return nil, status.Error(codes.Unauthenticated, err.Error())
 	}
 	endpoint.log.Debug("pinged", zap.Stringer("by", peerID.ID), zap.Stringer("srcAddr", p.Addr))
-	endpoint.service.wasPinged(time.Now(), peerID.ID, p.Addr.String())
+	endpoint.pingStats.wasPinged(time.Now(), peerID.ID, p.Addr.String())
 	return &pb.ContactPingResponse{}, nil
 }
+
+// WhenLastPinged returns last time someone pinged this node.
+func (stats *PingStats) WhenLastPinged() (when time.Time, who storj.NodeID, addr string) {
+	stats.mu.Lock()
+	defer stats.mu.Unlock()
+	return stats.lastPinged, stats.whoPingedNodeID, stats.whoPingedAddress
```

Comment:
it looks like this method `WhenLastPinged` isn't currently being used anywhere.  Is there a use case for this method to be used in the future? Im just thinking that if the node restarts for any reason, pingStats will get reset and these fields will be empty.  Not sure if that will affect anything

Reply:
It will be used by the dashboard to get the last contacted time. I'm also using it in the test